source/Octopus.Tentacle.Tests.Integration/ScriptServiceV2IntegrationTest.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut;
+using NUnit.Framework;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.Legacy;
+using Octopus.Tentacle.Contracts.ScriptServiceV2;
+using Octopus.Tentacle.Tests.Integration.Support;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    public class ScriptServiceV2IntegrationTest
+    {
+        [Test]
+        public async Task CanRunScript()
+        {
+            using IHalibutRuntime octopus = new HalibutRuntimeBuilder()
+                .WithServerCertificate(Support.Certificates.Server)
+                .WithMessageSerializer(s => s.WithLegacyContractSupport())
+                .Build();
+
+            var port = octopus.Listen();
+            octopus.Trust(Support.Certificates.TentaclePublicThumbprint);
+
+            using (var runningTentacle = await new PollingTentacleBuilder(port, Support.Certificates.ServerPublicThumbprint)
+                       .Build(CancellationToken.None))
+            {
+                var serviceEndPoint = new ServiceEndPoint(runningTentacle.ServiceUri, runningTentacle.Thumbprint);
+                var scriptServiceV2 = octopus.CreateClient<IScriptServiceV2>(serviceEndPoint);
+
+                var bashScript = "echo hello\nsleep 1";
+                var windowsScript = "echo hello\r\nStart-Sleep -Seconds 1";
+
+                var startScriptCommand = new StartScriptCommandV2Builder()
+                    .WithScriptBodyForCurrentOs(windowsScript, bashScript)
+                    .Build();
+
+                var response = scriptServiceV2.StartScript(startScriptCommand);
+
+                var (logs, finalResponse) = await RunUntilScriptCompletes(startScriptCommand, response, scriptServiceV2);
+
+                finalResponse.State.Should().Be(ProcessState.Complete);
+                finalResponse.ExitCode.Should().Be(0);
+
+                var allLogs = JoinLogs(logs);
+
+                allLogs.Should().Contain("hello");
+            }
+        }
+
+        private static string JoinLogs(List<ProcessOutput> logs)
+        {
+            return String.Join(" ", logs.Select(l => l.Text).ToArray());
+        }
+
+        [Test]
+        public async Task WhenTentacleRestartsWhileRunningAScript_TheExitCodeShouldBe_UnknownResultExitCode()
```

Comment:
This has failures, likely because of existing bugs in halibut.

https://build.octopushq.com/buildConfiguration/TeamFireAndMotion_OctopusTentacleNet48_SensibleDefaultsChainFullChain/7828186?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandPull+Request+Details=true&expandBuildTestsSection=true

To fix just retry the call after the tentacle is restarted

Reply:
When I rebased I lost the `nameof(IScriptServiceV2) + "Alpha"` in the capability service v2 so no retries are happening

Reply:
I fixed that issue up since that was failing a test which verified retries.

This is unrelated to that as this one doesn't use the tentacle client and instead calls the services directly.